File: src/main/java/com/partymakers/shareparty/data/persistence/party/impl/PartyRoomRepositoryImpl.java
package com.partymakers.shareparty.data.persistence.party.impl;

import com.partymakers.shareparty.data.persistence.party.PartyRoomPersistenceRepository;
import com.partymakers.shareparty.data.persistence.party.entity.PartyRoomEntity;
import com.partymakers.shareparty.domain.party.entity.PartyRoom;
import com.partymakers.shareparty.domain.party.port.PartyRoomRepository;
import lombok.RequiredArgsConstructor;

import java.util.Optional;
import java.util.stream.Collectors;

@RequiredArgsConstructor
public class PartyRoomRepositoryImpl implements PartyRoomRepository {

    private final PartyRoomPersistenceRepository repository;

    @Override
    public PartyRoom save(PartyRoom partyRoom) {
        return repository.save(PartyRoomEntity.Companion.toPersistence(partyRoom)).toDomain();
    }

    @Override
    public Optional<PartyRoom> findById(long id) {
        return repository.findById(id).flatMap(partyRoom -> Optional.of(partyRoom.toDomain()));
    }

    @Override
    public Iterable<PartyRoom> findAll() {
        return repository.findAll().stream().map(PartyRoomEntity::toDomain).collect(Collectors.toList());
    }
}

File: src/main/kotlin/com/partymakers/shareparty/data/persistence/party/impl/PartyRoomRepositoryImpl.kt
package com.partymakers.shareparty.data.persistence.party.impl

import com.partymakers.shareparty.data.persistence.party.PartyRoomPersistenceRepository
import com.partymakers.shareparty.data.persistence.party.entity.PartyRoomEntity
import com.partymakers.shareparty.domain.party.entity.PartyRoom
import com.partymakers.shareparty.domain.party.port.PartyRoomRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
class PartyRoomRepositoryImpl(
    private val repository: PartyRoomPersistenceRepository
) : PartyRoomRepository {

    override fun save(partyRoom: PartyRoom): PartyRoom =
        repository.save(PartyRoomEntity.toPersistence(partyRoom)).toDomain()

    override fun findById(id: Long): Optional<PartyRoom> =
        repository.findById(id).map { it.toDomain() }

    override fun findAll(): Iterable<PartyRoom> =
        repository.findAll().map { it.toDomain() }

    override fun deleteAll() {
        repository.deleteAll()
    }
} 